src/ngraph_builder.cc

```diff
@@ -1716,20 +1727,38 @@ tf::Status Builder::TranslateGraph(const std::vector<tf::TensorShape>& inputs,
       }
       auto lower_vec = ng_begin_const->get_vector<int>();
 
-      auto ng_size_const =
-          std::dynamic_pointer_cast<ng::op::Constant>(ng_size->second);
-      if (ng_size_const == nullptr) {
+      auto ng_end_const =
+          std::dynamic_pointer_cast<ng::op::Constant>(ng_end->second);
```

Comment:
We need to be storing this information as an attribute during the confirmation pass, rather than casting nodes to `Constant`.

Reply:
We'll still need to make this change. Take a look at `Reshape`'s confirmation function/builder case for one example of the pattern we need here.